src/snark_intf.ml

```diff
@@ -1157,6 +1211,56 @@ module type Run = sig
     val var : ('var, _) t -> 'var
   end
 
+  module Proof_system : sig
+    type ('a, 'inputs) t
+
+    val create :
+         ?proving_key:Proving_key.t
+      -> ?verification_key:Verification_key.t
+      -> ?proving_key_path:string
+      -> ?verification_key_path:string
+      -> ?handler:Request.Handler.t
+      -> exposing:(unit -> 'a, unit, 'computation, 'input) Data_spec.t
```

Comment:
I was thinking we should change the name of this to `public_input` to be more in sync with the usual nomenclature